File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i = 0
        j = len(heights) - 1
        maxA = 0
        while i < j:
            minH = min(heights[i], heights[j])
            currA = minH * (j-i)
            maxA = max(maxA, currA)
            if heights[i] < heights[j]:
                i += 1
            else:
                j -= 1
        return maxA

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-3.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        for(int i = 2; i < (int)cost.size(); i++){
            cost[i] += min(cost[i-1], cost[i-2]);
        }
        return min(cost[(int)cost.size()-2], cost[(int)cost.size()-1]);
    }
};
